lib/fsm/follow.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fsm
+
+import (
+	"context"
+	"time"
+
+	"github.com/gravitational/gravity/lib/storage"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+)
+
+// GetPlanFunc is a function that returns an operation plan.
+type GetPlanFunc func() (*storage.OperationPlan, error)
+
+// PlanEvent represents an operation plan event.
+type PlanEvent interface{}
```

Comment:
I'd also suggest to add an opaque unexported API to the interface so it could not be satisfied by an unrelated type:
```go
type PlanEvent interface {
	isPlanEvent()
}
```